File: app/src/main/java/com/mvvm/contactlist/db/AppDatabase.java
package com.mvvm.contactlist.db;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.sqlite.db.SupportSQLiteDatabase;

import com.mvvm.contactlist.db.dao.CategoryDao;
import com.mvvm.contactlist.db.dao.ContactDao;
import com.mvvm.contactlist.db.entities.CategoryEntity;
import com.mvvm.contactlist.db.entities.ContactEntity;


@Database(entities = {CategoryEntity.class, ContactEntity.class}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {
    private static final String DB_NAME = "contact_list.db";
    private static volatile AppDatabase database;

    public static AppDatabase getDatabase(final Context context) {
        if (database == null) {
            synchronized (AppDatabase.class) {
                if (database == null) {
                    database = Room.databaseBuilder(context.getApplicationContext(), AppDatabase.class, DB_NAME)
                            .addCallback(new Callback() {
                                @Override
                                public void onCreate(@NonNull SupportSQLiteDatabase db) {
                                    super.onCreate(db);
                                }
                            })
                            .build();
                }
            }
        }
        return database;
    }

    public abstract CategoryDao categoryDao();

    public abstract ContactDao contactDao();
}

File: app/src/main/java/com/mvvm/contactlist/db/AppDatabase.kt
package com.mvvm.contactlist.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room.databaseBuilder
import androidx.room.RoomDatabase
import com.mvvm.contactlist.db.dao.CategoryDao
import com.mvvm.contactlist.db.dao.ContactDao
import com.mvvm.contactlist.db.entities.CategoryEntity
import com.mvvm.contactlist.db.entities.ContactEntity

@Database(
    entities = [CategoryEntity::class, ContactEntity::class],
    version = 1,
    exportSchema = false
)
abstract class AppDatabase : RoomDatabase() {
    abstract fun categoryDao(): CategoryDao
    abstract fun contactDao(): ContactDao

    companion object {
        private const val DB_NAME = "contact_list.db"

        @Volatile
        private lateinit var database: AppDatabase

        fun getDatabase(context: Context): AppDatabase {
            synchronized(AppDatabase::class.java) {
                if (!::database.isInitialized) {
                    database = databaseBuilder(
                        context.applicationContext,
                        AppDatabase::class.java,
                        DB_NAME
                    ).build()
                }
            }
            return database
        }
    }
}